extensions/camel/camel-core/deployment/src/main/java/io/quarkus/camel/core/deployment/CamelRuntimeBuildItem.java

```diff
@@ -3,15 +3,17 @@
 import org.jboss.builder.item.SimpleBuildItem;
 
 import io.quarkus.camel.core.runtime.CamelRuntime;
+import io.quarkus.runtime.RuntimeValue;
 
 public final class CamelRuntimeBuildItem extends SimpleBuildItem {
-    private final CamelRuntime runtime;
 
-    public CamelRuntimeBuildItem(CamelRuntime runtime) {
+    private final RuntimeValue<CamelRuntime> runtime;
+
+    public CamelRuntimeBuildItem(RuntimeValue<CamelRuntime> runtime) {
         this.runtime = runtime;
     }
 
-    public CamelRuntime getRuntime() {
+    public RuntimeValue<CamelRuntime> getRuntime() {
```

Comment:
Don't you think you should unwrap it here? I think the idea is really to get the `CamelRuntime`.

Reply:
Not really. The point is to provide a proxy that could passed to a template method.